src/BsvUnifiedMerklePathView.tsx

```diff
@@ -23,26 +25,60 @@ interface MerklePathLeaf {
   height: number;
 }
 
+function calculateTheByteLengthOfBUMP(BUMP: BUMP) {
+  if (!BUMP || BUMP.path.length === 0) return 0;
+  let total = 4; // height
+  BUMP.path.forEach((level) => {
+    total += 1; // number of leaves
+    level.forEach((leaf) => {
+      total += 1; // offset
+      if (leaf.duplicate) {
+        total += 1;
+      } else {
+        total += 33;
+      }
+    });
+  });
+  return total;
+}
```

Comment:
Why we need so complex calculations here, instead of just doing it like in TSC format?

Reply:
When it displays large BUMPs the newline character is less frequently used. This has the effect that the format is more easily interpreted at a glance so that people can see how compact it is. Without this the JSON was immediately going off the screen which loses the effect.

Reply:
It's calculating the byte encoding not the JSON.